### PR TITLE
added random weight umap

### PR DESCRIPTION
**Pull request by xinyuchen-pku**

### CODE REVIEW

**Comment by newtontech** [comment]
## Kimi CLI Review

_Generated with `kimi --print` from PR metadata and patch diff. Review method: purpose-first, file/diff-based, severity-ordered findings, with tests/security/compatibility/CI impact checked._

## PR Review: added random weight umap

### 总体评估
此 PR 旨在为项目添加"随机初始化权重下的 UMAP 可视化"功能，通过引入 `random_weight` 参数，允许在预测/特征提取时使用随机初始化权重的模型，而非预训练权重。PR 包含代码修改、二进制模型文件、日志文件和可视化结果。

---

### 🔴 阻塞性问题 (Blocking Issues)

#### 1. **硬编码绝对路径** — `train/train_molclr/finetune.py`, `train/train_molclr/collect_data.py`
- `finetune.py` 中将数据路径从相对路径 `'data/perovskite-resplit/split_seed_0/train.csv'` 改为绝对路径 `/data/yhm/cxy_molclr/MolCLR/data/perovskite-resplit/split_seed_4/train.csv`，并新增 `perovskite`、`perovskite-new` 任务的绝对路径。
- `collect_data.py` 中 `base_dir` 被硬编码为 `/data/yhm/cxy_molclr/MolCLR/data/perovskite-resplit/split_seed_4/`。
- **风险**：这些路径是提交者本地环境特有的，其他用户无法运行。这严重破坏了代码的可移植性和可复现性。

#### 2. **占位符未替换** — `train/get_random_features.py`
```python
os.environ['UNIMOL_WEIGHT_DIR'] ='<your model path here>'
save_path = '<your model path here>'
```
- 脚本中包含未替换的占位符，直接运行会失败。虽然 README 中已有说明，但脚本本身不应包含无效值。

---

### 🟡 高风险问题 (High Risk)

#### 3. **提交了大量不应入仓的文件**
- **19 个 `__pycache__/*.pyc` 文件**：Python 字节码缓存不应提交到 Git，应在 `.gitignore` 中排除。
- **15 个日志文件** (`train/logs/uniqsar_*.log`)：运行时日志通常不应入仓，尤其包含大量本地路径信息。
- **2 个 `.pt` 二进制模型文件** (`addition_points_random_features_trained_model.pt`, `trainset_points_random_features_trained_model.pt`)：共约 620KB，这些似乎是运行产物而非源代码。
- **1 个 PNG 图片** (`visualize/umap_r.png`)：可视化结果。

> 55 个 changed files 中，真正需要代码审查的源文件仅约 6-7 个，其余都是运行产物或缓存。

#### 4. **随机权重初始化逻辑的实现方式** — `train/unimol_tools/models/nnmodel.py`
- PR 在 `nnmodel.py` 中新增了约 47 行代码来实现随机权重初始化，包括：
  - 使用 `torch.nn.init.normal_(mean=0.0, std=0.02)` 的方案（日志中可见）
  - 使用全局统计量 `mean=0.000291, std=0.042659` 的方案（日志中可见）
- 但 diff 被截断，无法完整审查该逻辑。从日志看，该功能似乎在多次迭代中经历了不同实现（日志显示 `nnmodel.py:137` 和 `nnmodel.py:159` 的不同版本），**最终实现的正确性和稳定性存疑**。

#### 5. **日志泄露本地系统信息**
- 日志文件中包含提交者的本地绝对路径 `/data/yhm/software/unimol_weights/weights/` 等，存在轻微信息泄露风险。

---

### 🟢 中低风险问题 (Medium/Low Risk)

#### 6. **README 更新不完整**
- README 新增了 `get_random_features.py` 的使用说明，但未说明需要手动替换占位符路径。

#### 7. **`collect_data.py` 的循环结构变更**
- 原代码遍历 `split_seed_nums` (0-4)，新代码硬编码为 `split_seed_4`。这不仅是路径问题，也改变了原有逻辑（从多 seed 集成变为单 seed），**这是否是有意为之？**

#### 8. **无测试覆盖**
- PR 未包含任何单元测试或集成测试来验证 `random_weight` 功能。

---

### 残留风险 / 测试缺口
- `random_weight=True` 时，模型是否完全绕过预训练权重加载？从日志看仍显示 `Loading pretrained weights from ...`，随后才显示权重重新初始化，**需要确认这是否会导致预训练权重先加载再覆盖**（影响效率和确定性）。
- 随机权重的可重复性：未设置随机种子，每次运行结果不可复现。
- `nnmodel.py` 的完整 diff 被截断，无法评估随机初始化对模型其他部分的影响。

---

### 结论
**此 PR 存在阻塞性问题，不建议直接合并。**

需要修复：
1. 移除所有绝对路径，恢复为相对路径或配置化；
2. 清理不应入仓的文件（`__pycache__`、日志、二进制 `.pt`、PNG），并更新 `.gitignore`；
3. 替换 `get_random_features.py` 中的占位符为合理默认值或环境变量读取；
4. 提供 `nnmodel.py` 的完整 diff 以供审查随机权重初始化逻辑；
5. 确认 `collect_data.py` 移除多 seed 循环是否是预期行为。

**Comment by newtontech** [comment]
Closed as superseded by a clean implementation on main in 37c991a. The useful random-weight Uni-Mol comparison path was extracted into docs/configs/examples/tests, while the generated model artifacts, logs, cache files, and images from this PR were intentionally not preserved.
